lib/errorUtils.js

```diff
@@ -25,6 +25,17 @@ function isSystem (err) {
   }
 }
 
+function handlePromiseError (err, reply) {
+  if (!(err instanceof Error)) {
```

Comment:
I think you should log `err` as it is.

Reply:
What do you mean with `log` ? We have to throw a `TypeError` otherwise the server won't exit (cause SystemError)

Reply:
I mean, you are not logging or putting the actual value of `err` anywhere, so a developer has no clue on where that was started.

Reply:
sure, it's logged [here](https://github.com/fastify/fastify/blob/master/fastify.js#L582)

Reply:
I’m lost. how can that be logged there, if the error is rewritten here? I would like to log the fact that you are rewriting this error.

Reply:
From the docs about `clientError` event

> If a client connection emits an 'error' event, it will be forwarded here

I could just do
```
 var newError = new TypeError(`Attempted to reject a promise with a non-error value from type '${typeof err}'`)
    reply.res.connection.emit('error', originalError)
throw newError 
```

Reply:
emitting `originalError` is wrong, because it is not a Error. We need to log `originalError`, attach the originalError to a property of the new `TypeError` with the original error itself

I don't think is correnct to emit error on `connection`, if you do it's better if you call `.destroy(err)`.

Reply:
Ok got it. Do we have any convention for error constructing?

 > I don't think is correnct to emit error on connection, if you do it's better if you call .destroy(err).

It's fine because `clientError` handler will clean up the ressource per definition. I also tried to use `destroy()` but I had strange failing tests and couldn't identify the root issue could you try it?

Reply:
I can't really help right now.

Reply:
@mcollina  @delvedor What're your thoughts about it? You think it's unsafe to close the connection via an emit? In my opinion, it feels clean because all client errors are handled in one central place.